src/parser/flags.ts

```diff
@@ -12,15 +12,15 @@ export function build(
   defaults: Partial<OptionFlag<string>>,
 ): Definition<string>
 export function build<T>(defaults: Partial<OptionFlag<T>>): Definition<T> {
```

Comment:
unrelated to this PR, but back our previous conversation about GlobalFlags...I wonder if it would good to have an `O` generic property that reflects the `options` type.  

so `integer` could pass in its mix/max, dir/file could do `exists` etc